storage/google/cloud/storage/notification.py

```diff
@@ -28,10 +28,10 @@
 NONE_PAYLOAD_FORMAT = 'NONE'
 
 _TOPIC_REF_FMT = '//pubsub.googleapis.com/projects/{}/topics/{}'
-_PROJECT_PATTERN = r'(?P<project>[a-z][a-z0-9-]{4,28}[a-z0-9])'
+_PROJECT_PATTERN = r'(?P<project>\A[a-z][a-z0-9-]{4,28}[a-z0-9]\Z)'
```

Comment:
@chemelnucfin Why not use `^` and `$`?

Reply:
probably better in this case so it doesn't match \n?

Reply:
Which is probably better? Newlines won't get matched by default

Reply:
but if someone accidentally wrote one?

Reply:
I guess it's not that big a deal for me to change it.  But just putting my thoughts out there.

Reply:
changed